packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -786,19 +786,17 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
   void showOnScreen([RenderObject child]) {
     // Logic duplicated in [_RenderSingleChildViewport.showOnScreen].
     if (child != null) {
-      // TODO(goderbauer): Don't scroll if it is already visible.
-      // TODO(goderbauer): Don't guess if we need to align at leading or trailing edge.
-      // Move viewport the smallest distance to bring [child] on screen.
       final double leadingEdgeOffset = getOffsetToReveal(child, 0.0);
       final double trailingEdgeOffset = getOffsetToReveal(child, 1.0);
       final double currentOffset = offset.pixels;
-      // TODO(goderbauer): Don't scroll if that puts us outside of viewport's bounds.
-      if ((currentOffset - leadingEdgeOffset).abs() < (currentOffset - trailingEdgeOffset).abs()) {
+
+      if (currentOffset > leadingEdgeOffset && currentOffset > trailingEdgeOffset) {
```

Comment:
can you assert `leadingEdgeOffset <= trailingEdgeOffset`?

Reply:
No, but I can guarantee it the other way around: `leadingEdgeOffset >= trailingEdgeOffset` :)

Reply:
(really? that's surprising)

in that case, assert that then simplify the expressions

Reply:
To get a list item to the leading edge (as opposed to the trailing edge) you have to scroll further, therefore `leadingEdgeOffset >= trailingEdgeOffset`, no?

Will simplify expression.

Reply:
In a regular vertically scrolling list, the leading edge is above the trailing edge, so you get it first (lower scroll offset) than the trailing edge (higher scroll offset) unless I'm missing something?

Reply:
To ensure that we talk about the same thing I have drawn up two diagrams: Both show a long scrollable list (on the left) that scrolls through the viewport window (on the right in yellow). 

In the first picture, the list item in question is shown at the leading edge of the viewport. I'd argue that the scrollOffset here is 600, which is called the `leadingEdgeOffset` in the code.

![img-1935](https://user-images.githubusercontent.com/1227763/40322528-8b808f72-5ce7-11e8-9af1-a3300efe1226.JPG)

Now, let's move the list item in question to the trailing edge of the viewport. I'd argue that in this picture the scrollOffset is 350, which is called the `trailingEdgeOffset` in the code.

![img-1937](https://user-images.githubusercontent.com/1227763/40322623-c93eb53c-5ce7-11e8-921d-838b6553ab1b.JPG)

So, now I would argue that 600 > 350. And therefore (even though this is by now means a full mathematically proof), it seems that `leadingEdgeOffset >= trailingEdgeOffset` holds.

Reply:
ah, yup, i was missing something. :-)